### PR TITLE
POC/Provisioning: Add id mapping utility

### PR DESCRIPTION
**Pull request by ryantxu**
This PR updates our haphazard approach to mapping names so we at least have a function.  The details on the function may still require iteration, but I think best to start with *something*

See: https://docs.google.com/document/d/1uyF2bJcgEnFL7RqoXDMsn_-PqNylkrXBmrNdKEQWMHI/edit?tab=t.fy14i8p5xa8t#heading=h.2x8418tdyfs4

### CODE REVIEW

File: pkg/registry/apis/provisioning/resources/parser.go [hunk 1]
```diff
@@ -164,15 +157,9 @@ func (r *Parser) Parse(ctx context.Context, logger *slog.Logger, info *repositor
 		Timestamp: nil, // ???&info.Modified.Time,
 	})
 
-	// When name is missing use the file path as the k8s name
-	if obj.GetName() == "" {
```
Comment (ryantxu) [inline pkg/registry/apis/provisioning/resources/parser.go:168]: ^^^ the main change

File: pkg/registry/apis/provisioning/resources/parser.go [hunk 2]
```diff
@@ -43,6 +42,7 @@ func (f *ParserFactory) GetParser(repo repository.Repository) (*Parser, error) {
 		repo:   config,
 		client: client,
 		kinds:  kinds,
+		mapper: NamesFromFileName,
```
Comment (ryantxu) [inline pkg/registry/apis/provisioning/resources/parser.go:45]: ^^^ what we are currently using, but not what I think we should use

File: pkg/registry/apis/provisioning/resources/idmapper_test.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+func TestIDMapper(t *testing.T) {
+	obj := &unstructured.Unstructured{}
+
+	dashName, folderName := NamesFromHashedRepoPath("xyz", "path/to/folder/dashboard.json", obj)
+	require.Equal(t, "dashboard-B60LqDCLR6FC", dashName)
+	require.Equal(t, "folder-aObw9Q2jRwTI", folderName)
+
+	name1, f1 := NamesFromHashedRepoPath("xyz", "path/to/folder.json", obj)
```
Comment (ryantxu) [inline pkg/registry/apis/provisioning/resources/idmapper_test.go:17]: to support setting folder title/labels etc, we can have a yaml|json file with the same prefix as a folder represent a folder?

File: pkg/registry/apis/provisioning/resources/idmapper.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	anno := obj.GetAnnotations()
+	if anno != nil {
+		folderName = anno[utils.AnnoKeyFolder]
+	}
+	return
+}
+
+// Will pick a name based on the hashed repository and path
+func NamesFromHashedRepoPath(repo string, fpath string, _ metav1.Object) (objectName string, folderName string) {
+	// remove the extension
```
Comment (ryantxu) [inline pkg/registry/apis/provisioning/resources/idmapper.go:50]: this implies we should error if a ".yaml" and ".json" exist with the same prefix 🤔 

alternatively we keep the extension in the hash, but do not display it.  Means changing from json/yaml is a new resoruce, but I think that is OK

**Review by Proximyst — APPROVED** [review]
love this!

File: pkg/registry/apis/provisioning/resources/idmapper_test.go (hunk 3 above)
Comment (Proximyst) [reply pkg/registry/apis/provisioning/resources/idmapper_test.go:17]: I think this is the simplest, best idea to solve the problem. It'd be nice not lock us into this solution, though, as we could support metadata on the folder itself (e.g. in S3) as well...

**Review by MissingRoberto — APPROVED** [review]

**Pull request merged**